rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/serialization/NodeSourceConfigurationParser.java

```diff
@@ -60,7 +62,7 @@ public NodeSourceConfiguration parseNodeSourceConfiguration(String json) {
         try {
             jsonObject = parseJSON(json).isObject();
         } catch (JSONException e) {
-            throw new IllegalArgumentException("The imported node source is not a valid JSON file", e);
+            throw new IllegalArgumentException(JSON_FILE_ERROR_MESSAGE, e);
         }
 
         try {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "nodesRecoverable" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)